### PR TITLE
use validation api for cli local deploys

### PR DESCRIPTION
**Pull request by dan-barrett**
Closes https://gitlab.com/architect-io/architect-cli/-/issues/49
Closes https://gitlab.com/architect-io/architect-hub/-/issues/120

### CODE REVIEW

**Comment by dan-barrett** [comment]
@tjhiggins hold off on this, looks like a test failing

**Comment by dan-barrett** [comment]
@tjhiggins okay, good to go again.

File: src/commands/deploy.ts [hunk 1]
```diff
@@ -230,4 +242,34 @@ export default class Deploy extends Command {
       await this.runRemote();
     }
   }
+
+  async validate_graph(graph: DependencyGraph): Promise<void> {
+    cli.action.start(chalk.blue('Validating deployment'));
+    const { data: validation_results } = await this.app.api.post<ValidationResult[]>(`/graph/validation`, graph);
```
Comment (tjhiggins) [inline src/commands/deploy.ts:248]: Lets make this non-blocking in a non-200 response. Someone without internet should still be able to do a local deploy.

Comment (dan-barrett) [reply src/commands/deploy.ts]: Done 👍

**Pull request merged**